rules/prefer-flat-map.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const MESSAGE_ID = 'preferFlatMap';
+
+const isFlat = node => {
+	return (
+		node.type === 'CallExpression' &&
+		node.callee.type === 'MemberExpression' &&
+		node.callee.property.type === 'Identifier' &&
+		node.callee.property.name === 'flat'
+	);
+};
+
+const isMap = node => {
+	return (
+		node.type === 'CallExpression' &&
+		node.callee.type === 'MemberExpression' &&
+		node.callee.property.type === 'Identifier' &&
+		node.callee.property.name === 'map'
+	);
+};
```

Comment:
These two methods are almost identical. Could be DRY'd up into a `isMethodNamed('map')` and moved to util.